### PR TITLE
Updated the `ResNetFRCNN` to return as its class labels the label strings

### PR DESCRIPTION
**Pull request by Purg**

### CODE REVIEW

**Comment by Purg** [comment]
@schencej PTAL

**Review by joshanderson-kw — APPROVED** [review]

File: smqtk_detection/impls/detect_image_objects/resnet_frcnn.py [hunk 1]
```diff
@@ -17,61 +19,77 @@
...
+        model = self.get_model()
 
-        formatted_dets = []       # AxisAlignedBoundingBox detections to return
-
-        img_tensors = [self.model_loader(img) for img in img_iter]
+        img_list = list(img_iter)
+        img_tensors = [self.model_loader(img) for img in img_list]
 
         if self.use_cuda:
             img_tensors = [tensor.cuda() for tensor in img_tensors]
```
Comment (schencej) [inline smqtk_detection/impls/detect_image_objects/resnet_frcnn.py:95]: This will cause an error if `use_cuda` is true but there is no cuda device available because the model weights and input will be on different devices. Perhaps you could set `use_cuda` to false in `get_model()` when no device is found.

**Review by schencej — APPROVED** [review]
Other than my one comment, LGTM. Thanks for updating and adding tests!

File: smqtk_detection/impls/detect_image_objects/resnet_frcnn.py [hunk 2]
```diff
@@ -17,61 +19,77 @@
...
-        else:
-            self.use_cuda = False
+        # Set to None for lazy loading later.
+        self.model: torch.nn.Module = None  # type: ignore
 
+        # The model already has normalization and resizing baked into the
+        # layers.
         self.model_loader = transforms.Compose([
-            transforms.ToPILImage(),
+            # transforms.ToPILImage(),
```
Comment (Purg) [inline smqtk_detection/impls/detect_image_objects/resnet_frcnn.py:55]: Oops, dead code. I will remove.

File: smqtk_detection/impls/detect_image_objects/resnet_frcnn.py (hunk 1 above)
Comment (Purg) [reply smqtk_detection/impls/detect_image_objects/resnet_frcnn.py:95]: I did not re-assign the `self.cuda` parameter thinking that it would be better to not perturb what was input so that the output `get_config` is reflective of what was input, while the logic would be idempotent (modulo different environments) in its interpretation of those parameters. Of course, now that I think about it, it might just be better to straight up error if you set use-cuda and there is no device to use...

A second error that you just highlighted is that we should not be using `.cuda` and instead use `to(device=...)`.
